[v10.3.x] Chore: Improve domain validation for Google OAuth - Backport 83229 to v10.3.x (#83725)
grafana/grafana: docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md

You may specify a domain to be passed as `hd` query parameter accepted by Google's
OAuth 2.0 authentication API. Refer to Google's OAuth [documentation](https://developers.google.com/identity/openid-connect/openid-connect#hd-param).

{{% admonition type="note" %}}

> Since Grafana 10.3.0, the `hd` parameter retrieved from Google ID token is also used to determine the user's hosted domain. The Google Oauth `allowed_domains` configuration option is used to restrict access to users from a specific domain. If the `allowed_domains` configuration option is set, the `hd` parameter from the Google ID token must match the `allowed_domains` configuration option. If the `hd` parameter from the Google ID token does not match the `allowed_domains` configuration option, the user is denied access.

Reviewer: @linoman , I'm confused by "Since Grafana 10.3.0" - this is for 10.3, correct? Are there any other versions where this feature is available? I'm wondering if we can simply start the sentence with "The `hd` parameter..." and not mention a version number. Thoughts?

> When an account does not belong to a google workspace, the hd claim will not be available.

Reviewer: ```suggestion
When an account does not belong to a Google Workspace, the `hd` claim is not available.
```

Author: Changed as requested.
